src/include/sof/audio/component.h

```diff
@@ -238,7 +238,9 @@ enum {
 		  (uint32_t)((pcd)->cpu_delta_peak))
 
 #define comp_perf_avg_info(pcd, comp_p)					\
-	comp_info(comp_p, "perf comp_copy cpu avg %u (current peak %u)",\
+	comp_info(comp_p, "perf comp_copy samples %u period %u cpu avg %u peak %u",\
+		  (uint32_t)((comp_p)->frames),            \
+		  (uint32_t)((comp_p)->period),			    \
```

Comment:
`frames` and `period` are defined as `uint32_t` in `src/include/sof/audio/component.h`. Why are they cast to `uint32_t`? Casts hide bugs from the compiler, we want fewer casts not more.

Reply:
BTW renaming `comp_p` to `comp_dev_p` would save a lot of time to people new to this code (which involves an unreasonable number of macro indirections for "functional programming in C")